### PR TITLE
Handle openat/open relative paths 

### PR DESCRIPTION
**Pull request by infiniteregrets**
Following changes have been made in the pull request:
- if open is called with a relative path, we just propagate that call to libc otherwise the call goes to agent 
- if openat is called with an absolute path or `AT_FDCWD` then we send the call to open otherwise we check our hashmap for existing descriptors if we dont have a dirfd we make the libc call 
- formatted python files with black
- a test to verify read-write file operation

### CODE REVIEW

File: tests/python-e2e/ops.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import unittest
+
+TEXT = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
+
+
+class FileOpsTest(unittest.TestCase):
+    def test_read_write_family(self):
```
Comment (aviramha) [inline tests/python-e2e/ops.py:7]: how does this test ensure it's running remotely?

File: mirrord-agent/src/file.rs [hunk 2]
```diff
@@ -17,11 +17,15 @@ use tracing::{debug, error};
 
 use crate::{error::AgentError, runtime::get_container_pid, sniffer::DEFAULT_RUNTIME, PeerID};
 
-// TODO: To help with `openat`, instead of `HashMap<_, File>` this should be
-// `HashMap<_, RemoteFile`>, where `RemoteFile` is a struct that holds both the `File` + `PathBuf`.
+#[derive(Debug)]
+pub struct RemoteFile {
```
Comment (aviramha) [inline mirrord-agent/src/file.rs:21]: why do we need it..?

Comment (aviramha) [reply mirrord-agent/src/file.rs:21]: when openat is used, we just need a different object for it, but that object doesn't need a path either..

**Review by aviramha — CHANGES_REQUESTED** [review]

File: mirrord-agent/src/file.rs [hunk 3]
```diff
@@ -36,10 +47,28 @@ impl FileManager {
...
+                    .map_err(|err| {
+                        error!(
+                            "FileManager::open -> Failed to get metadata for file {:#?}",
+                            err
+                        );
+                        err
+                    })
+                    .map(|metadata| {
+                        debug!("FileManager::open -> Got metadata for file {:#?}", metadata);
+                        if metadata.is_dir() {
```
Comment (eyalb181) [inline mirrord-agent/src/file.rs:66]: Stylewise, I would suggest
```suggestion
                        let value = if metadata.is_dir() { RemoteFile::Directory(path)} else {RemoteFile::File(file)};
                        self.open_files.insert(fd, value);
```

File: mirrord-agent/src/file.rs [hunk 4]
```diff
@@ -17,11 +18,21 @@ use tracing::{debug, error};
 
 use crate::{error::AgentError, runtime::get_container_pid, sniffer::DEFAULT_RUNTIME, PeerID};
 
-// TODO: To help with `openat`, instead of `HashMap<_, File>` this should be
-// `HashMap<_, RemoteFile`>, where `RemoteFile` is a struct that holds both the `File` + `PathBuf`.
+/// Unqiue file descriptor generator
+pub(crate) fn get_fd() -> usize {
```
Comment (eyalb181) [inline mirrord-agent/src/file.rs:22]: Might be clearer to call this generate_fd, get implies you're retrieving an existing value

File: mirrord-agent/src/file.rs [hunk 5]
```diff
@@ -113,65 +192,87 @@ impl FileManager {
...
+                );
+
+                WriteFileResponse {
+                    written_amount: write_amount,
+                }
+            })
+        } else {
+            return Err(ResponseError::NotFound);
+        }
+        .map_err(|fail| {
```
Comment (eyalb181) [inline mirrord-agent/src/file.rs:266]: Why isn't this inside the if clause (rather than chained to the whole if/else)?

**Review by eyalb181 — APPROVED** [review]

**Review by aviramha — APPROVED** [review]

**Pull request merged**